doc/docker.md

```diff
@@ -1,16 +1,102 @@
+```sh
+# To log in with OIDC
+OIDC_*
+# To interact with AWS account IAM config
+AWS_DATA_ACCOUNT_ID
+AWS_COMPUTE_ACCOUNT_ID
+```
+
+Unless you're testing the slack feature, just disable it with this line:
+```sh
+SLACK_API_TOKEN=disabled
+```
+
+For more details of environment variable settings, refer to: [Environment Variables Reference](environment.md).
+
+## AWS setup
+
+Control Panel needs AWS credentials to change S3 bucket IAM permissions, according to user requests. When this runs in the cluster it can get them using kube2iam. But when you run locally you'll need to provide AWS credentials that give access to the mojanalytics account. Typically you'll use configure to use your Landing Account User and use AssumeRole to the mojanalytics account.
+
+You can test what AWS Account is currently configured on your command-line, like this:
+
+```sh
+$ pip install boto3
+$ python -c "import boto3; print(boto3.client('sts').get_caller_identity()['Arn'])"
+arn:aws:sts::593291632749:assumed-role/restricted-admin-data/botocore-session-1590188888
+```
+
+As an AP developer, if you don't have a Landing AWS Account user account yet, follow the steps here: https://github.com/ministryofjustice/analytical-platform-iam#user-creation Make sure you've completed *all* of the 'First login' section.
+
+Now you will create an Access Key and save it in a file on your machine. In the AWS Console, go to IAM | Users | Security credentials | Access keys | Create access key. Now while you've still got the values for AWS Access Key ID and AWS Secret Access Key on the screen, in your terminal run `aws configure` and input those values as answers to the first two questions. Set 'Default region name' to `eu-west-1` and 'Default output format' to blank. This writes those credentials as your `[default]` profile to ~/.aws/credentials.
+
+Now edit ~/.aws/credentials and add this second 'profile':
+
+```ini
+[data]
+role_arn = arn:aws:iam::593291632749:role/restricted-admin-data
+source_profile = default
+```
+
+You can configure to use this 'data' profile using:
+
+```sh
+export AWS_PROFILE=data
+```
+
+and boto3 or awscli commands will access the mojanalytics/data AWS Account, by using your Landing Account creds and switching.
+
+boto3 looks in the [usual places for AWS credentials](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#configuring-credentials). 
+
+## kubernetes config setup
+
+You need to have a Kubernetes key at ~/.kube/controlpanel. The dev cluster is fine to use.
+
+To do this:
+
+1. Browse to: https://kuberos.services.dev.mojanalytics.xyz/
+2. Login with your GitHub creds and Auth0 MFA code for dev
+3. Look for the second line, where it says "Save **this file** as ~/.kube/config". Download the file linked at "this file"
+4. Move the file to `~/.kube/controlpanel` e.g. `mv ~/Downloads/kubecfg.yaml ~/.kube/controlpanel`
+
+## Run
+
 ```sh
 docker-compose up
 ```
```

Comment:
I get:

```
$ docker-compose up
ERROR: Couldn't find env file: /home/ntoll/src/moj/analytics-platform-control-panel/.env
```

Ahh... right, I need the `.env` file. See comment at the start.

Reply:
Hopefully the instructions explain how to create it now